src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -739,73 +777,93 @@ DatabaseShardImp::import(Database& source)
             JLOG(j_.error()) << "invalid source database";
             return;
         }
+    }
```

Comment:
This block does not need a lock. Since we know this function is only interested in importing the node store, I recommend removing the source param and using app_.getNodeStore() where applicable. That would eliminate this block entirely, the comment can remain.

Reply:
Done - though I did keep the `init_` check. It's been moved to the leading function (`DatabaseShardImp::importDatabase`).